src/sbvr-api/errors.ts

```diff
@@ -6,9 +6,368 @@ export class PermissionParsingError extends TypedError {}
 export class UnsupportedMethodError extends TypedError {}
 export class SqlCompilationError extends TypedError {}
 export class SbvrValidationError extends TypedError {}
-export class InternalRequestError extends TypedError {}
 
 export class TranslationError extends TypedError {}
 export class ParsingError extends TypedError {}
-export class BadRequestError extends TypedError {}
-export class ForbiddenError extends TypedError {}
+
+export interface AnyObject {
+	[index: string]: any
+}
+
+export class HttpError extends TypedError {
+	constructor(
+		public status: number,
+		error: string | Error = '',
+		public body?: string | AnyObject
+	) {
+		super(error)
+	}
+
+	getResponseBody() {
+		return this.body !== undefined ?
+			this.body :
+			this.message
+	}
+}
+
+export class BadRequestError extends HttpError {
+	constructor(
+		error: string | Error = '',
```

Comment:
You could change all of the subclasses to use
```typescript
error?: string | Error
```
as `HttpError` will handle the defaulting

Reply:
Nice :+1:
Testing, rebasing, squashing & merging.